pkg/object/websocketserver/spec.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package websocketserver
+
+import (
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"strings"
+)
+
+type (
+	// Spec describes the HTTPServer.
+	Spec struct {
+		Port    uint16 `yaml:"port" jsonschema:"required,minimum=1"`
+		HTTPS   bool   `yaml:"https" jsonschema:"required"`
```

Comment:
HTTPS? Should it be `WSS` or `EnableTLS`?

Reply:
Actually, this service can enable TLS in two places, one is the HTTP server it starts with, the other is the backend WebSocket's. 
* This `HTTPS` is for the original HTTP server this proxy works on. (Before protocol switching)
* The backend WebSocket service's TLS configure can be enabled when it starts with `wss` in URL.